src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -642,12 +647,29 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                         // If `falseReg` is zero, then move the first operand of `intrinEmbMask` in the
                         // destination using /Z.
 
-                        assert(targetReg != embMaskOp2Reg);
-                        GetEmitter()->emitIns_R_R_R(INS_sve_movprfx, emitSize, targetReg, maskReg, embMaskOp1Reg, opt);
+                        switch (intrinEmbMask.id)
+                        {
+                            case NI_Sve_CreateBreakPropagateMask:
+                                assert(targetReg != maskReg);
+                                if (targetReg != embMaskOp2Reg)
```

Comment:
nit: This check is technically unnecessary, the `/* canSkip */ true` will no-op if `targetReg == embMaskOp2Reg`

Reply:
Yes, it should no-op as the `IsRedundantMov` would return `true`. We have several places where we need to do this kind of cleanup/simplification.